### PR TITLE
[build] Fix ordering issues with ExposeExperimentalFeatures compiler switch

### PR DESCRIPTION
**Pull request by CodeBlanch**
## Changes

* Fixes the order of things so `EXPOSE_EXPERIMENTAL_FEATURES` constant is set AFTER the version resolution logic fires

### CODE REVIEW

File: build/Common.prod.props [hunk 1]
```diff
@@ -59,6 +59,19 @@
     <Message Importance="high" Text="**AssemblyVersionDebug** TargetFramework: $(TargetFramework), MinVerVersion: $(MinVerVersion), MinVerMajor: $(MinVerMajor), MinVerMinor: $(MinVerMinor), MinVerPatch: $(MinVerPatch), MinVerPreRelease: $(MinVerPreRelease), BuildNumber: $(BuildNumber), FileVersion: $(FileVersion), ExposeExperimentalFeatures: $(ExposeExperimentalFeatures)" />
   </Target>
 
+  <Target Name="ResolveExposeExperimentalFeatures" AfterTargets="MinVer">
```
Comment (utpilla) [inline build/Common.prod.props:62]: Shouldn't this be run after `AssemblyVersionTarget`?

```suggestion
  <Target Name="ResolveExposeExperimentalFeatures" AfterTargets="AssemblyVersionTarget">
```

Comment (CodeBlanch) [reply build/Common.prod.props:62]: I had that at first but `AssemblyVersionTarget` gets skipped for non-publish builds and we want this section to always run.

Comment (utpilla) [reply build/Common.prod.props:62]: When publishing builds, is the ordering guaranteed? We don't want this target to be evaluated before `AssemblyVersionTarget`.

Comment (CodeBlanch) [reply build/Common.prod.props]: Not 100% sure about that, but I changed it so it now has a `DependsOnTargets="AssemblyVersionTarget"`

**Review by utpilla — APPROVED** [review]

**Review by reyang — APPROVED** [review]

**Pull request merged**